### ISSUE TITLE
Howdy

### ISSUE THREAD

**Issue by johndbro1** [jonm/SillyMUD#97]
I don't know how to say this via a message - but this is Loki.  I meant to say hi sooner, but got distracted by the holidays, and then forgot until earlier today.

I'm super-excited you're trying to refactor it.   It should be an epic adventure in its own right, heh.

good luck!  Let me know if there's anything I can do to help, other than code, can't really commit to that right now.

**Comment by jonm** [comment]
Hi Loki! Glad you found us, and I'm glad you are supportive of the project. :) I played a character called Hobbes on the original Silly; wrote a C client to have it try to play as a robot for a while. The "Hobbes Express" could take you various places in the game if you followed him from the Midgaard reception.

We've been hacking away at the MUD off and on--kind of feels like restoring a classic car on weekends. It's currently running at `silly.jjmoore.net` on port 4000 if you want to poke around a bit for old times' sake.

We've encountered some things in the code we didn't entirely understand, or couldn't remember how parts of the game were supposed to work. If we run across any of those, perhaps we can @-mention you in an issue to see if you remember. :)

**Comment by johndbro1** [comment]
> We've encountered some things in the code we didn't entirely understand,
> or couldn't remember how parts of the game were supposed to work. If we run
> across any of those, perhaps we can @-mention you in an issue to see if you
> remember. :)

yeah, I bet.  There were a few things I don't remember myself, but I'm
happy to help if I can.

John

**Comment by jonm** [comment]
And it goes without saying that should you feel inclined to pitch in with code that would naturally be welcome, but please don't feel any pressure to commit time to it. We have been picking it up and putting it down ourselves; it's really about spending whatever time we find enjoyable doing it.

**Comment by JerrySievert** [comment]
I can also help "interpret" a few things :)

**Comment by JerrySievert** [comment]
as an aside, I may still have some of the original world sitting around, and maybe a copy of `phoenix`, the followup.

will try to spend some time looking (obviously been planning on that, given that I found this today)

**Comment by comcast-jonm** [comment]
@JerrySievert great! I haven't had a chance to push on this for a while, but it still sits around in the back of my mind. :) What character(s) were you on Silly?

**Comment by JerrySievert** [comment]
@comcast-jonm see: http://muds.wikia.com/wiki/SillyMUD for "Sievert" - "DM".

also wrote a huge chunk of Epic (where we were multiplexing the socket for much more than 64 connections), and continued on with Phoenix.

started looking for the laptop that I think has some old code, but had to give up for a few hours.

**Comment by JerrySievert** [comment]
having @johndbro1 and I on this thread should give you a major leg up if you decide to revisit :)

**Comment by JerrySievert** [comment]
just found this:

```
#13
DM's Playhouse~
   As you look around, you see a cluttered room full of various
books on psychology and artificial intelligence.  Resting in one
corner is a black leather couch like those found in psychiatrist's
offices.  Next to it is an open notepad and a pen.  In another corner
you notice a small iced tea maker half full.
~
0 8712 0
E
notepad~
On the notepad, the following is written:

   Today's subjects:
     1) holding back emotions
     2) stifeling creativity

   Changes required:
     1) full rewrite of decision making fuzzy logic
     2) new Natural Language Parser - old fried?

~
S
```

**Comment by osusan** [comment]
wow... a lot of familiar names! I had a zillion characters (all morts); Petrol, Osu, Titanic, Turin, etc. I Went over to RO after Silly shut down and ended up helping out a few muds that Silly spawned (1 recently shutdown TheUnderworld). I used to love grouping with Iuz, Conner's mort, and always enjoyed a good chat with Stranger, Gambreezi, and on the rare occasion Xenakis when he would bless us :) It was a special time for me at SillyMud and UFL, so many good memories and laughs.

Great work on the refactor as well. Got it to boot nicely under Ubuntu, with the only issues relating to the install of the dependencies. I've been so used to using the "set command" as imm, that I forgot about @ (which has no help file). Re the game itself, any way to load zones on boot vs have them load once entered? 

Cheers, Osu

**Comment by jonm** [comment]
Hey @johndbro1 and @JerrySievert : I'm thinking we should properly credit you and as many of the Silly devs/contributors as we can. Currently `lib/credits` doesn't really list you, which seems like a big oversight. If you'd like to open PRs with how you'd like to be credited, happy to merge them. Also, if you know of other original Silly devs we should also invite (even if just for that purpose!), please tag them in here.

Or if you just want to tell me how you'd like to appear, I can just do it if the PR dance is a hassle. :)

**Comment by JerrySievert** [comment]
I see myself and John: https://github.com/jonm/SillyMUD/blob/main/lib/credits#L4

**Comment by jonm** [comment]
Duh. I totally missed it. Thanks!

**Comment by jonm** [comment]
Hey, anyone remember what figurines were/did? I never ran into one when I was playing back in the day. When I boot the MUD, one of the messages indicates it can't find the `figurines` file. It's missing from our distribution, and I don't know if this is intentional (because figurines were created via gameplay?) or accidentally (there were a fixed number of pre-created figurines). Seems like you could use a figurine to summon a mob as a follower for a short period of time?

**Comment by JerrySievert** [comment]
figurines were an item that could turn into a mob like a familiar for some period of time.  if memory serves, the figurines file was the mapping between item number and mob number.

**Comment by JerrySievert** [comment]
```
  while (fgets(buf, 254, f1)) {
    sscanf(buf, "%d %d\n", &figurine[ fig_count ].obj,
           &figurine[ fig_count ].mob);
    fig_count++;
    if (fig_count >= 255)
      fig_count = 254;
  }
```

wow, I can't believe I actually remembered that correctly.
